GomokuClient/packages/gomoku-core/src/hooks/useGameSession.ts

```diff
@@ -1,21 +1,25 @@
 import { useEffect } from "react";
 import { TileColor, useTiles } from "@/hooks/useTiles";
-import { typedStorage } from "@/shared/lib/utils";
+import { getDefaultHeaders } from "@/shared/lib/utils";
 import { useSignalRConnection } from "@/context";
+import { SwaggerServices, SwaggerTypes } from "@/api";
+import { genToArray } from "@/utils/gen.utility";
+import { useQuery } from "@tanstack/react-query";
 
-export const useJoinGame = (gameID: string) => {
-  const { tiles, winner, addTile, lastTile } = useTiles(gameID);
+export const useGameSession = (gameID: string) => {
+  const { tiles, setTiles, winner, addTile, lastTile, setLastTile } =
+    useTiles();
   const { hubProxy, isConnected, registerEventHandlers } =
     useSignalRConnection();
 
+  const { data: gameHistory } = useGameHistory(gameID);
+
   useEffect(() => {
-    const previousGameID = typedStorage.getItem("currentGameID");
-    if (previousGameID && previousGameID !== gameID) {
-      typedStorage.removeItem(`gameBoard_${previousGameID}`);
-      typedStorage.removeItem(`nextTurn_${previousGameID}`);
+    if (gameHistory) {
+      setTiles(genToArray(gameHistory.gen));
+      setLastTile(gameHistory.movesHistory[gameHistory.movesCount]);
     }
-    typedStorage.setItem("currentGameID", gameID);
-  }, [gameID]);
+  }, [gameHistory]);
```

Comment:
useEffect has missing dependencies: 'setLastTile' and 'setTiles'. Either include them or remove the dependency array